src/constraints/storage.jl

```diff
@@ -60,8 +60,11 @@ function add_storage_constraints!(connection, model, variables, expressions, con
                         @constraint(
                             model,
                             var_storage_level[row.id] ==
-                            previous_level + profile_agg * row.storage_inflows + incoming_flow -
-                            outgoing_flow,
+                            (
+                                1 - row.storage_loss_from_stored_energy
+                            )^length(row.time_block_start:row.time_block_end) * previous_level +
```

Comment:
I suggest pre-computing `(1 - r)^length` before the `@constraint`. Something like this:

```julia
computed_storage_loss_coef = 1.0
if row.storage_loss_from_stored_energy > 0.0
    duration = row.time_block_end - row.time_block_start + 1
    computed_storage_loss_coef = (1 - row.storage_loss_from_stored_energy)^duration
end
```